chain-storage-sqlite-old/src/lib.rs

```diff
@@ -292,5 +334,6 @@ mod tests {
         put_get();
         nth_ancestor();
         iterate_range();
+        simultaneous_read_write()
```

Comment:
Why must these functions be called from single `#[test]` entry point, rather than each one annotated with `#[test]`?

Reply:
There are two things:

- cargo runs tests in multiple threads
- sqlite manages shared cache somewhere in the background and doesn't expose any context to a user, there is only one "global context"

Because of that concurrent tests are not isolated. There are three possible ways to deal with that:

- Limit the number of threads cargo uses to run tests. This is very inconvenient when running tests for the whole repo and needs additional documentation and attention from the user who wants to run tests.
- Put all tests under one function to prevent them from being run in parallel. This is what I did.
- Create test databases with URLs like `file:{db_name}?mode=memory&cache=shared` so that there are different in-memory databases. Actually, this may be a better solution.

Reply:
I see. We can continue like this, while documenting the reason in comments.
But to improve test granularity it's better to use separate databases.